### PR TITLE
Fix rich text utils backspace not clearing the block type at the start.

### PR DESCRIPTION
**Pull request by jvaill**
**Summary**

When backspacing at the start of the first block using rich text utils, the block type should be cleared.

Example of me frantically trying to clear the first block type **without** this PR by pressing backspace at the beginning of the document:

clear-first-block-type

Followed by an example of me clearing the first block type **with** this PR:

clear-first-block-type-working

**Test Plan**
- Pull master.
- Open the rich text example.
- Create a list.
- Put your cursor at the first offset of the first block, backspace to reset the block type, notice that it doesn't work.
- Pull this branch.
- Open the rich text example.
- Create a list.
- Put your cursor at the first offset of the first block, backspace to reset the block type, rejoice in joy as the list item disappears.
- Celebrate. 🎉

### CODE REVIEW

**Comment by jvaill** [comment]
^ Looks like Travis might need a retry?

File: src/model/modifier/__tests__/RichTextEditorUtil-test.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe('RichTextEditorUtil', () => {
 
       // Remove the current text from the blockquote.
       const resetBlockquote = DraftModifier.removeRange(
-        editorState.getCurrentContent(),
+        contentState,
```
Comment (flarnie) [inline src/model/modifier/__tests__/RichTextEditorUtil-test.js]: Nice - thanks for fixing that.

File: src/model/modifier/__tests__/RichTextEditorUtil-test.js [hunk 2]
```diff
@@ -82,6 +82,27 @@ describe('RichTextEditorUtil', () => {
       expect(lastBlockNow.getText()).toBe('');
     });
 
+    it('resets the current block type at the start of the first block', () => {
```
Comment (flarnie) [inline src/model/modifier/__tests__/RichTextEditorUtil-test.js]: Thanks for writing a test!

testing_i_love_testing

**Comment by flarnie** [comment]
I'm going to try and resolve the conflicts, ensure CI passes, and merge this.

**Comment by jvaill** [comment]
Woot! Awesome to see this get merged. Thank you!

**Pull request closed without merging**